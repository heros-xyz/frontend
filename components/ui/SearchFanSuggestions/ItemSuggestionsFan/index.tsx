import { Box, Flex, Text, Image } from "@chakra-ui/react";
import { IFanInfo } from "@/types/athlete/types";

interface IProps {
  item: IFanInfo;
  onClickItem?: (item: IFanInfo) => void;
}

const ItemSuggestionsFan: React.FC<IProps> = ({ item, onClickItem }) => {
  return (
    <Box
      borderBottom="1px"
      borderColor="gray.100"
      py="2"
      cursor={"pointer"}
      onClick={() => onClickItem && onClickItem(item)}
    >
      <Flex>
        <Box
          mr="2.5"
          borderRadius="full"
          bg="success.default"
          overflow="hidden"
        >
          <Image
            src={item.avatar}
            alt="heros item"
            width={{ base: "40px", xl: "50px" }}
            height={{ base: "40px", xl: "50px" }}
            objectFit="cover"
          />
        </Box>
        <Box display="flex" justifyContent="center" flexDirection="column">
          <Text
            color="primary"
            fontSize={{ base: "xs", xl: "md" }}
            mb={{ xl: 1 }}
            fontWeight={"700"}
            lineHeight="140%"
          >
            {item?.fullName}
          </Text>
        </Box>
      </Flex>
    </Box>
  );
};

export default ItemSuggestionsFan;
