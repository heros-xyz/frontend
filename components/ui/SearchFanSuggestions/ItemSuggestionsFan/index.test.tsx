import { render } from "@testing-library/react";
import ItemSuggestionsFan from ".";

it("renders List unchanged", () => {
  const { container } = render(
    <ItemSuggestionsFan
      item={{
        avatar: "",
        fullName: "Matt Prior",
        description: "----",
        createdAt: "2023-11-11",
        id: "1",
        email: "abcxyz@gmail.com",
      }}
    />
  );
  expect(container).toMatchSnapshot();
});
