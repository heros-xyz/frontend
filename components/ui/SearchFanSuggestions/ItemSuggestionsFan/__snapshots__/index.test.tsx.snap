// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List unchanged 1`] = `
<div>
  <div
    class="css-kcc1xl"
  >
    <div
      class="css-k008qs"
    >
      <div
        class="css-haoc2a"
      >
        <img
          alt="heros item"
          class="chakra-image__placeholder css-k6jbwm"
          src="https://via.placeholder.com/50"
        />
      </div>
      <div
        class="css-zupw4a"
      >
        <p
          class="chakra-text css-wwubfu"
        >
          Matt Prior
        </p>
      </div>
    </div>
  </div>
</div>
`;
